EnterInfoBox.cs

```diff
@@ -48,15 +48,14 @@ private void InfoBoxConfirm_Click(object sender, EventArgs e)
         {
             // Checks whether input values are numbers
             string numberPatternRegex = @"(^\d*\.?\d*[1-9]+\d*$)|(^[1-9]+\d*\.\d*$)";
-            bool isPriceValid = Regex.IsMatch(TextBoxGoalItemPrice.Text, numberPatternRegex);
-            bool isSavingsValid = Regex.IsMatch(TextBoxSavings.Text, numberPatternRegex);
-            bool isSalaryValid = Regex.IsMatch(TextBoxMonthlySalary.Text, numberPatternRegex);
-            if (!isPriceValid || !isSavingsValid || !isSalaryValid) 
+            string error = "Not a number";
+            string isPriceValid = error.ErrorMessageIfNotMatchesRegex(numberPatternRegex, TextBoxGoalItemPrice.Text);
```

Comment:
field `isPriceValid`, as well as the following ones, indicate that this is a boolean, when in fact it's a string. Could be better off as something like `priceErrorMessage`